### ISSUE TITLE
Add initial integration tests

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#159]
Right now we have some JS unit tests, but no integration tests at all. We should start adding integration tests using `@testing-library/react` and perhaps packages like `@testing-library/user-event`

**Comment by swissspidy** [comment]
Here's the documentation for React Testing Library:

https://testing-library.com/docs/react-testing-library/intro
https://testing-library.com/docs/react-testing-library/setup
https://testing-library.com/docs/react-testing-library/api
https://testing-library.com/docs/react-testing-library/faq
https://testing-library.com/docs/react-testing-library/cheatsheet

Relevant companion libraries / packages:

https://github.com/testing-library/user-event
https://github.com/testing-library/jest-dom
https://testing-library.com/docs/ecosystem-eslint-plugin-testing-library
https://testing-library.com/docs/ecosystem-eslint-plugin-jest-dom

**Comment by swissspidy** [comment]
We should probably change code coverage analysis to

* run coverage also for PRs
* add back code coverage [PR comments](https://docs.codecov.io/docs/pull-request-comments)

**Comment by dvoytenko** [comment]
@swissspidy, I tried to construct a simple test that looked like this:

```
import { renderHook, act } from '@testing-library/react-hooks';

import FrameElement from '../../components/canvas/frameElement';

describe.only('test', () => {
  it('should render a frame element', () => {
    const element = {
      id: '1',
      type: 'text',
    };
    renderHook(() => (<FrameElement element={element} />));
  });
});
```

And I'm seeing the parse error on one of the icons.

<details>
<summary>SyntaxError: Unexpected token &lt;</summary>
 ● Test suite failed to run

    Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html

    Details:

    /Users/dvoytenko/DNI/web-stories-wp/assets/src/edit-story/elements/background/icon.svg:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 29 29"><path fill="currentColor" fill-rule="evenodd" d="M18.179 11.0613L19.0124 11.0599C19.929 11.0583 20.6803 11.8071 20.6818 12.7237L20.6958 21.0571C20.6973 21.9737 19.9486 22.725 19.0319 22.7265L9.03192 22.7433C8.11526 22.7448 7.364 21.9961 7.36247 21.0794L7.3485 12.7461C7.34697 11.8294 8.09571 11.0782 9.01238 11.0766L9.84571 11.0752L9.84291 9.40857C9.83906 7.10857 11.7026 5.23878 14.0026 5.23492C16.3026 5.23107 18.1724 7.09461 18.1762 9.3946L18.179 11.0613ZM14.0054 6.90157C12.6221 6.90389 11.5073 8.02242 11.5096 9.40575L11.5124 11.0724L16.5124 11.064L16.5096 9.39737C16.5073 8.01404 15.3888 6.89925 14.0054 6.90157ZM9.02917 21.0766L9.01521 12.7433L19.0152 12.7265L19.0292 21.0599L9.02917 21.0766ZM15.6888 16.8988C15.6903 17.8154 14.9416 18.5667 14.0249 18.5682C13.1083 

    SyntaxError: Unexpected token <

      23 |  * Internal dependencies
      24 |  */
    > 25 | import { ReactComponent as Icon } from './icon.svg';
         | ^
      26 | 
      27 | function BackgroundIcon() {
      28 |   const alt = __('Background element', 'web-stories');

      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:537:17)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:579:25)
      at Object.<anonymous> (assets/src/edit-story/elements/background/icon.js:25:1)
</details>

Any advice?

**Comment by swissspidy** [comment]
@dvoytenko check out how @barklund fixed this in #383
